src/blocks/battery.rs

```diff
@@ -86,11 +89,27 @@ impl PowerSupplyDevice {
 
 impl BatteryDevice for PowerSupplyDevice {
     fn is_available(&self) -> bool {
+        // the problem with hid devices is, that in case they are 
+        // not plugged in the whole power_supply subdirectory doesn't exist
+        if self.device.starts_with("hid")
+        && !self.device_path.exists() {
```

Comment:
I think this check is redundant?

Reply:
device is just the device string from the config file and we don't want to say "not available" for devices that are normal batteries. Otherwise those would be assumed to be not available when they are actually configured wrong

Reply:
I'm not sure I follow. If the base directory does not exist, it means the battery is not available, doesn't matter whether it's hid or not. And actually, that was the only indicator before https://github.com/greshake/i3status-rust/commit/24f432fb67e5ba3cadddf5084b60c15e392f5e44.

Reply:
Well but normal Batteries like BAT0 never change their sysfs path. So it would be better to show the user that they entered the wrong path instead of saying "device is missing". I guess either version makes sense but my reasoning was that in this case throwing an error is the right thing because a BATASDF that doesn't exists is not _really_ missing rather than misconfigured.

Reply:
With HID devices it's a bit different because they remove the whole device path when they are unplugged (which is kind of the reason for PR #1455)

Reply:
> Well but normal Batteries like BAT0 never change their sysfs path

They can also be plugged/unplugged (I think so, I don't have the necessary hardware to test myself), so it's not true. And I don't see how this check affects anything - if `device_path` doesn't exist, the function will return `false` in _any_ case.

Reply:
You're right that makes sense.